### PR TITLE
Minor: Update docstrings and comments to aggregate code

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

N/A

# Rationale for this change

While working on https://github.com/apache/arrow-datafusion/pull/4488 I had to remind myself of this information several times, so I added comments (largely selfishly for my future self)

# What changes are included in this PR?

Update docstrings

# Are these changes tested?

N/A
# Are there any user-facing changes?

Docs only

### CODE REVIEW

File: datafusion/core/src/physical_plan/aggregates/hash.rs [hunk 1]
```diff
@@ -504,7 +510,7 @@ fn create_batch_from_map(
         })
         .collect::<Result<Vec<_>>>()?;
 
-    // add state / evaluated arrays
+    // next, output  aggregates: either intermediate state or final output
```
Comment (jackwener) [inline datafusion/core/src/physical_plan/aggregates/hash.rs:513]:
```suggestion
    // next, output aggregates: either intermediate state or final output
```

File: datafusion/expr/src/accumulator.rs [hunk 2]
```diff
@@ -31,28 +31,48 @@ use std::fmt::Debug;
 /// * update its state from multiple accumulators' states via `merge_batch`
 /// * compute the final value from its internal state via `evaluate`
 pub trait Accumulator: Send + Sync + Debug {
-    /// Returns the state of the accumulator at the end of the accumulation.
-    /// in the case of an average on which we track `sum` and `n`, this function should return a vector
-    /// of two values, sum and n.
+    /// Returns the partal intermediate state of the accumulator. This
```
Comment (jackwener) [inline datafusion/expr/src/accumulator.rs:34]:
```suggestion
    /// Returns the partial intermediate state of the accumulator. This
```

File: datafusion/expr/src/accumulator.rs [hunk 3]
```diff
@@ -31,28 +31,48 @@ use std::fmt::Debug;
...
         Err(DataFusionError::Internal(
             "Retract should be implemented for aggregate functions when used with custom window frame queries".to_string()
         ))
     }
 
-    /// updates the accumulator's state from a vector of states.
+    /// Updates the accumulator's state from an `Array` containing one
+    /// or more intermediate values.
+    ///
+    /// The `states` array passed was formed by cancatenating the
```
Comment (jackwener) [inline datafusion/expr/src/accumulator.rs:68]:
```suggestion
    /// The `states` array passed was formed by concatenating the
```

**Review by jackwener — APPROVED** [review]
I read them carefully.
Look good to me except typos

**Comment by alamb** [comment]
Thanks @jackwener

**Pull request merged**